openbas-api/src/main/java/io/openbas/rest/inject/service/InjectAssistantService.java

```diff
@@ -0,0 +1,432 @@
+package io.openbas.rest.inject.service;
+
+import io.openbas.database.model.*;
+import io.openbas.database.repository.AttackPatternRepository;
+import io.openbas.database.repository.InjectRepository;
+import io.openbas.injectors.manual.ManualContract;
+import io.openbas.rest.exception.ElementNotFoundException;
+import io.openbas.rest.inject.form.InjectAssistantInput;
+import io.openbas.rest.injector_contract.InjectorContractService;
+import io.openbas.service.AssetGroupService;
+import io.openbas.service.EndpointService;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.Query;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@RequiredArgsConstructor
+@Validated
+public class InjectAssistantService {
+
+  private final AssetGroupService assetGroupService;
+  private final EndpointService endpointService;
+  private final InjectorContractService injectorContractService;
+  private final AttackPatternRepository attackPatternRepository;
+  private final InjectRepository injectRepository;
+
+  private InjectorContract manualInjectorContract = null;
+
+  @PersistenceContext private EntityManager entityManager;
+
+  public List<Inject> generateInjectsForScenario(Scenario scenario, InjectAssistantInput input) {
```

Comment:
*Nitpick (non-blocker) :* Javadoc on top of the public methods (or even better on the private methods too) would be nice